timbermill-client/src/main/java/com/datorama/timbermill/pipe/TimbermillServerOutputPipe.java

```diff
@@ -0,0 +1,170 @@
+package com.datorama.timbermill.pipe;
+
+import com.datorama.timbermill.unit.Event;
+import com.datorama.timbermill.unit.EventsWrapper;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.*;
+
+public class TimbermillServerOutputPipe implements EventOutputPipe {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TimbermillServerOutputPipe.class);
+    private URL timbermillServerUrl;
+    private List<Event> buffer;
+    private Thread thread;
+    private int maxEventsBatchSize;
+    private long maxSecondsBeforeBatchTimeout;
+
+
+    private TimbermillServerOutputPipe() {
+    }
+
+    private TimbermillServerOutputPipe(Builder builder){
+        if (builder.timbermillServerUrl == null){
+            throw new RuntimeException("Must enclose the Timbermill server URL");
+        }
+        try {
+            timbermillServerUrl = new URL(builder.timbermillServerUrl);
+        } catch (MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+        maxEventsBatchSize = builder.maxEventsBatchSize;
+        maxSecondsBeforeBatchTimeout = builder.maxSecondsBeforeBatchTimeout;
+        buffer = Collections.synchronizedList(new LinkedList<>());
+    }
+
+    private void startThread() {
+        thread = new Thread(() -> {
+            do {
+                try {
+                    List<Event> eventsToSend = createEventsToSend();
```

Comment:
check if we can turn this to a blocking queue so we don't iterate over emty buffer over and over again